File: sample-videochat-webrtc/QBRTCChatSample/Cells/UserTableViewCell/UserTableViewCell.m
//
//  UserTableViewCell.m
//  QBRTCChatSemple
//
//  Created by Andrey Ivanov on 11.12.14.
//  Copyright (c) 2014 QuickBlox Team. All rights reserved.
//

#import "UserTableViewCell.h"
#import "CornerView.h"

@interface UserTableViewCell()

@property (weak, nonatomic) IBOutlet CornerView *colorMarker;
@property (weak, nonatomic) IBOutlet UILabel *userDescriptionLabel;

@end

@implementation UserTableViewCell

- (void)setColorMarkerText:(NSString *)text andColor:(UIColor *)color {
    
    self.colorMarker.bgColor = color;
    self.colorMarker.title = text;
}

- (void)setUserDescription:(NSString *)userDescription {
    
    if (![_userDescription isEqualToString:userDescription]) {

        _userDescription = userDescription;
        self.userDescriptionLabel.text = userDescription;
    }
}

@end

File: sample-chat-swift/sample-chat-swift/Classes/Views/UserTableViewCell.swift
//
//  UserTableViewCell.swift
//  sample-chat-swift
//
//  Created by Injoit on 1/28/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

class UserTableViewCell: UITableViewCell {
    @IBOutlet fileprivate weak var colorMarker: CornerView!
    @IBOutlet fileprivate weak var userDescriptionLabel: UILabel!
    
    var user: QBUUser?
    var dialogID = ""
    
    var userDescription:String! {
        didSet {
            userDescriptionLabel.text = userDescription
        }
    }
    
    func setupColorMarker(_ color: UIColor){
        colorMarker.backgroundColor = color
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        let markerColor = colorMarker.backgroundColor
        super.setSelected(selected, animated: animated)
        colorMarker.backgroundColor = markerColor
    }
    
    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
        let markerColor = colorMarker.backgroundColor
        super.setHighlighted(highlighted, animated: animated)
        colorMarker.backgroundColor = markerColor
    }
}
